### DISCUSSION TITLE
Has anyone successfully booted the Haxophone from Crowd Supply using a Raspberry Pi Zero?

### DISCUSSION THREAD

**Discussion by hortense667** [cardonabits/haxo-hw#51] [Q&A] [answered]
Hi,

I received the Haxophone from Crowd Supply.
I immediately transferred the software image and booted up the Haxophone, but there is no sound.
I used the Raspberry Pi Imager for the software image transfer, and performed it as follows:

Raspberry Pi Device: Raspberry Pi Zero
OS: User Custom
I specified the haxophone.img downloaded from the following site:
https://bit.ly/haxo-rpiz
Storage: MicroSD drive
When I connected the Raspberry Pi to HDMI and powered it up, it displayed the screen attached.
It stops at the Login screen.
Please let me know if there was anything wrong with my approach.

Satoshi Endo

![_2023-12-22 14 40 52](https://github.com/cardonabits/haxo-hw/assets/25192979/dc3c02ec-e209-482d-a10e-2a1525d16624)

**Comment by jcard0na** [comment]
Hi Satoshi,

What is the size of the image you downloaded?  We run into quota issues where some people were unable to download the image.  I uploaded a new image a few hours ago that is zipped for a faster download and to mitigate the quota limitations.

You can fetch it from the [bit.ly link](https://bit.ly/haxo-rpiz) as well as this [private backup link](http://94.143.139.181/haxophone-rpiz.img.zip).

The image is zipped, so you will have to unzip it before flashing.  There is also a checksum included in the zipfile to verify the image.  This is how to use it:

```
#! /bin/sh
#
wget http://94.143.139.181/haxophone-rpiz.img.zip
unzip haxophone-rpiz.img.zip
sha256sum -c haxophone-rpiz.img.sha256
```
and this is the expected output
```
--2023-12-22 12:33:00--  http://94.143.139.181/haxophone-rpiz.img.zip
Connecting to 94.143.139.181:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 3462777973 (3.2G) [application/zip]
Saving to: ‘haxophone-rpiz.img.zip’

haxophone-rpiz.img.zip       100%[==============================================>]   3.22G  8.97MB/s    in 6m 48s  

2023-12-22 12:39:48 (8.10 MB/s) - ‘haxophone-rpiz.img.zip’ saved [3462777973/3462777973]
inflating: haxophone-rpiz.img      
inflating: haxophone-rpiz.img.sha256  
haxophone-rpiz.img: OK
```
If you get the same output you will know that `haxophone-rpiz.img` is a valid image that you can flash.

Let me know if that worked!
Cheers,

**Reply by hortense667** [reply]
Hi,

Thank you for your quick response.

The size of the software image I downloaded yesterday is 14.8 GB (15,931,539,456 bytes). The haxophone-epiz.img that I received this time was also 14.8 GB (15,931,539,456 bytes), which means it was the same size.

Despite this, I was able to boot it up this time! I was even able to play it!

However, after playing it a couple of times, it stopped booting up again. 
It seems to be unstable in its operation. I am attaching a screenshot of the screen at that time. 
Should I log in and perform some operations? 
What would be the Login ID and Password in that case? Or should these be set when creating the MicroSD card? 
If you have any advice regarding this, please let me know.

Satoshi Endo
![2023-12-23 00 41 45](https://github.com/cardonabits/haxo-hw/assets/25192979/fca56a4e-db31-4add-a39a-fb5fd120499b)

**Comment by jcard0na** [comment]
Hi Satoshi,

I asked about the size because before the image was uncompressed (16 GB),
but as of today, the image is zipped to approx 3.3 GB.  Once uncompressed,
both images are the same size.

Regarding the reliability issue, that might be an SD card problem, it is
not an error I have observed.  Is it recoverable, or after it starts
failing it no longer works?  What if you re-flash the image?

If you want to login, the default user is pi and password haxophone.

Let me know what you discover!

Best,

**Reply by hortense667** [reply]
Hi,

I replaced the MicroSD card with a different 32GB one and loaded the software image onto it. 
However, it won't boot up. I haven't been able to boot it up even once since then. 
I am attaching a photo of the screen again for this attempt

Tomorrow, I would like to try again with another different MicroSD card. What does the swap file error mean?
Additionally, I have one Raspberry Pi 3. Therefore, I would like to try using that as well tomorrow.

Satoshi Endo

![2023-12-23 02 15 00](https://github.com/cardonabits/haxo-hw/assets/25192979/9cc81f38-ae82-4ba2-8c09-e9405a06399e)

**Comment by jcard0na** [comment]
The error message suggests to execute a systemctl status command to see
more details, can you do that?

**Reply by hortense667** [reply]
This is the result of the execution!
![2023-12-23 12 48 49](https://github.com/cardonabits/haxo-hw/assets/25192979/59e6c6e4-b583-48fe-8c8b-3fd3ac95c65a)

**Comment by jcard0na** [comment, marked as answer]
Thank you, Satoshi.

TLDR;  The swapfile error is not preventing the boot to complete, or the haxophone to work.  Details follow...

OK, I think I understand now.  The haxophone image is configured to boot as read-only, using `raspi-config` as recommended by the Raspberry Pi documentation [1](https://www.raspberrypi.com/documentation/computers/configuration.html#overlay-file-system-2) [2](https://learn.adafruit.com/read-only-raspberry-pi/overview).  As such, there is no swapfile.  On some systems, a swapfile is used to supplement RAM by using disk storage, but that is not available on the Raspberry Pi in read-only mode.  The swapfile is enabled on boot by this systemd service, dphys-swapfile, and it fails when it finds the SD card to be read-only.
`raspi-config` should have disabled that service when configuring the SD card as read-only, and apparently it does not.  However  that error is benign:  it is just informing you that the swapfile could not be created.  But the system is fully operational, and the `haxo` service should be running.  Have you tried to play?

At least this is what happens on my setup:

I get the same swapfile error...
```
pi@haxophone:~$ systemctl status dphys-swapfile.service --no-pager -l
● dphys-swapfile.service - dphys-swapfile - set up, mount/unmount, and delete a swap file
     Loaded: loaded (8;;file://haxophone/lib/systemd/system/dphys-swapfile.service/lib/systemd/system/dphys-swapfile.service8;;; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Thu 2023-12-21 20:35:34 GMT; 11min ago
       Docs: 8;;man:dphys-swapfile(8)man:dphys-swapfile(8)8;;
    Process: 380 ExecStart=/sbin/dphys-swapfile setup (code=exited, status=0/SUCCESS)
    Process: 438 ExecStart=/sbin/dphys-swapfile swapon (code=exited, status=255/EXCEPTION)
   Main PID: 438 (code=exited, status=255/EXCEPTION)
        CPU: 1.126s

Dec 21 20:35:27 haxophone systemd[1]: Starting dphys-swapfile - set up, mount/unmount, and delete a swap file...
Dec 21 20:35:29 haxophone dphys-swapfile[380]: want /var/swap=100MByte, checking existing: keeping it
Dec 21 20:35:34 haxophone dphys-swapfile[439]: swapon: /var/swap: swapon failed: Invalid argument
Dec 21 20:35:34 haxophone systemd[1]: dphys-swapfile.service: Main process exited, code=exited, status=255/EXCEPTION
Dec 21 20:35:34 haxophone systemd[1]: dphys-swapfile.service: Failed with result 'exit-code'.
Dec 21 20:35:34 haxophone systemd[1]: Failed to start dphys-swapfile - set up, mount/unmount, and delete a swap file.
Dec 21 20:35:34 haxophone systemd[1]: dphys-swapfile.service: Consumed 1.126s CPU time.
```

...but haxo service is running:
```
pi@haxophone:~$ systemctl status haxo.service --no-pager -l                                                                                                                                                 
● haxo.service - haxophone
     Loaded: loaded (8;;file://haxophone/etc/systemd/system/haxo.service/etc/systemd/system/haxo.service8;;; enabled; vendor preset: enabled)
     Active: active (running) since Thu 2023-12-21 20:35:30 GMT; 20min ago
    Process: 449 ExecStartPre=/usr/local/bin/create_midi_gadget.sh (code=exited, status=0/SUCCESS)
   Main PID: 563 (haxo001)
      Tasks: 3 (limit: 395)
        CPU: 14min 4.074s
     CGroup: /system.slice/haxo.service
             └─563 /usr/local/bin/haxo001 --notemap-file /usr/share/haxo/notemap.json --prog-number 66 --sf2-file /usr/share/sounds/sf2/TimGM6mb.sf2

Dec 21 20:35:31 haxophone systemd[1]: Started haxophone.
Dec 21 20:35:37 haxophone haxo001[563]: fluidsynth: warning: SDL2 not initialized, SDL2 audio driver won't be usable
Dec 21 20:35:37 haxophone haxo001[563]: [2023-12-21T20:35:37Z INFO  haxo001::alsa] Found alsa card MAX98357A
Dec 21 20:35:47 haxophone haxo001[563]: Synth created
Dec 21 20:35:47 haxophone haxo001[563]: [2023-12-21T20:35:47Z INFO  haxo001::midi]
Dec 21 20:35:47 haxophone haxo001[563]:     Available output ports:
Dec 21 20:35:47 haxophone haxo001[563]: [2023-12-21T20:35:47Z INFO  haxo001::midi] 0: Midi Through:Midi Through Port-0 14:0
Dec 21 20:35:47 haxophone haxo001[563]: [2023-12-21T20:35:47Z INFO  haxo001::midi] 1: f_midi:f_midi 24:0
Dec 21 20:35:47 haxophone haxo001[563]: [2023-12-21T20:35:47Z INFO  haxo001::midi] Picked the last one
Dec 21 20:35:47 haxophone haxo001[563]: Starting haxophone...
```

And everything works.  So if you do not get audio, I would say that the swapfile error is not the problem here.

Let me know if you find anything else.  We have to get you playing!

**Reply by hortense667** [reply]
Hi,

I am able to play it now! 
There was a problem with the speaker. 
I misunderstood it as a system issue because of the error that was occurring. 
My apologies. 
Thank you for your patience and assistance over this extended period.
Thank you for the wonderful electronic gadget!

Satoshi Endo

**Comment by jcard0na** [comment]
I’m so happy to hear that, thank you so much.

Best,

**Reply by hortense667** [reply]
It was my misunderstanding that caused you trouble. My apologies.

Satoshi Endo

**Comment by jcard0na** [comment]
None needed!  That error message is misleading and should not be there.
I’ll see if I can manually disable that service if I ever make another
release.  Cheers

**Reply by hortense667** [reply]
https://twitter.com/hortense667/status/1738608758102929836

**Reply by jcard0na** [reply]
So nice!  Thanks for sharing!
Just to provide additional details to:

> 作っているのはポルトガルの人らしい。

I'm originally from Spain, lived all my professional life in the US, but I conceived and designed the Haxophone in Portugal where I currently reside.  The fabrication was done between two places, China for the main PCB and SMD assembly, and Portugal the assembly of all the through hole components, mechanical parts, quality assurance and boxing.  And now the Haxophone is played in Japan.  To my knowledge, it is the second time in history that a Haxophone has been played there (I was in Japan this summer with my Haxophone, so I suppose I was first 😉)

All the best,

**Reply by hortense667** [reply]
My apologies.

**Reply by jcard0na** [reply]
> My apologies.

Oh! Do not apologize, please 😛 !  Nothing you have done or said is wrong!  I was just giving you additional context as you seemed interested in the origins of the Haxopone.

I'm really happy that I was able to help, that you now have a playable instrument and that you shared your playing with the world!  This was an instrument that I developed during the dark days of Covid, in the loneliness of my lab.  Every additional person that finds joy playing the Haxophone makes me very happy.  The world needs more joy!

So, thank you very much! 🤗  And Merry Christmas! 🎅

**Reply by hortense667** [reply]
Merry Christmas!!
